### ISSUE TITLE
Algorithm checklist 1

### ISSUE THREAD

**Issue by ajaysub110** [SforAiDl/genrl#6]
As discusses, let's start off with some popular algorithms in each domain not necessarily in this order. Feel free to add and update this when necessary.

1. **Deep RL**
- [x] Vanilla Policy Gradient
- [x] PPO1
- [x] DDPG
- [ ] DQN (all variants as options for a single model)
- [x] SAC
- [x] TD3

2. **Classical RL**
- [x] UCB-1
- [x] Thompson Sampling
- [x] SARSA (with eligibility traces)

**Comment by ajaysub110** [comment]
I'll work on HER

**Comment by Sharad24** [comment]
It would be great if we can log results from algorithm implementations. That way, we dont have to run them again later on

**Comment by ajaysub110** [comment]
Yeah. Can you create a spreadsheet where we add results (mean and variance of rewards) on different environments?

**Comment by Sharad24** [comment]
We can probably have a markdown file with a table, etc inside dir `deeprl/` no?

**Comment by ajaysub110** [comment]
Yeah makes sense. Should we also create a json/yml file for hyperparameters?

**Comment by Sharad24** [comment]
Yeah. We can add these details to the Contributing Guidelines I think

**Comment by ajaysub110** [comment]
Ok, will add that once these files have been made

**Comment by Sharad24** [comment]
@mehulrastogi Can you work on adding any evolutionary algorithms?

**Comment by mehulrastogi** [comment]
For sure sharad!

**Comment by Sharad24** [comment]
@mehulrastogi Could you please create an issue on the same so that we can discuss basic structure, etc beforehand?

**Comment by mehulrastogi** [comment]
Okay so there are many Evolutionary Algorithms. I gues you want to cover the basic and the popular ones right?
so those can be listed as follows:-

-  Basic ES  (https://openai.com/blog/evolution-strategies/)
- NEAT
- HyperNEAT
- Many more are there. You can look at [link](https://en.wikipedia.org/wiki/Neuroevolution)

**Comment by Sharad24** [comment]
We can start off with NEAT, ES algos [here](https://github.com/uber-research/deep-neuroevolution) and HyperNEAT (in any order). These look like the main ones for now, we can keep adding stuff later on. What do you think?

**Comment by Sharad24** [comment]
Of course, we should focus on the ones which are reproducible+benchmarkable first, otherwise we might just be stuck with code no ones gonna look at

**Comment by mehulrastogi** [comment]
> Of course, we should focus on the ones which are reproducible+benchmarkable first, otherwise we might just be stuck with code no ones gonna look at

I think to focus on NEAT and HyperNEAT first. Of course you can find the implementation for NEAT for python [here](https://neat-python.readthedocs.io/en/latest/). If you want NEAT implementation in PyTorch you can find them [here](https://github.com/uber-research/PyTorch-NEAT). After replicating the benchmark for NEAT for your tasks you will want to focus on HyperNEAT and then the Deep Neuroevolution algorithms by UBER.  In my opinion 
Random Search -> Basic ES -> NEAT -> HyperNEAT -> DeepNeuroevolution.

**Comment by Sharad24** [comment]
Ok, thanks for the info! 

I was hoping you could take some of these on? If not, thats ok, we will come back to it later after getting some core details of deeprl sorted.

**Comment by Sharad24** [comment]
Closing this in favour of #101 as thats the only one left
